staging/src/k8s.io/component-base/logs/kube-log-runner/README.md

```diff
@@ -15,6 +15,65 @@ Why do we need this?
 
+## Flags
+
+- -flush-interval
+  The `-flush-interval` flag is a duration flag that specifies how frequently the log 
+  file content is flushed to disk. The default value is 0, meaning no periodic flushing. 
+  If set to a non-zero value, the log file is flushed at the specified interval.
+
+  Type: Duration
+  
+  Default: 0 (flushing disabled)
+  
+  Usage: When set to non-zero , the log file is flushed every specified interval. 
+  While this may not be necessary on Linux, on Windows, it ensures that recent log 
+  entries are written to disk in near real-time, which is particularly useful for 
+  users who need to monitor logs as they are generated without delay.
+
+- -log-file-size
+  The `-log-file-size` flag is an optional string flag that sets a size limit for 
+  the log file, triggering automatic log rotation when the specified size is reached.
+  This is especially useful in production environments where the log file may become
+  too large to view effectively.
+  Beware that rotation can happen at arbitrary points in the byte stream emitted by the command.
+  This can lead to splitting a log entry into two parts, with the first part in one file
+  and the second part in the next file.
+
+  Type: String (expects a value in Resource.Quantity format, such as 10M or 500K)
+
+  Default: "0" (disabled, no automatic rotation of log files)
+
+  Usage: When set to a positive value, the log file will rotate upon reaching the specified 
+  size limit. The current log file’s contents will be saved to a backup file, and a new log 
+  file will be created at the path specified by the `-log-file` flag, ready for future log entries.
+
+  Backup File Naming Convention:
+    `<original-file-name>-<timestamp><file-extension>`.
+    * `<original-file-name>`: The name of the original log file, without the file extension.
+    * `<timestamp>`: A timestamp is added to each backup file’s name to uniquely identify it
+    based on the time it was created. The timestamp follows the format "20060102-150405".
+    For example, a backup created on June 2, 2006, at 3:04:05 PM would include this timestamp.
+    * `<file-extension>`: The original file’s extension (e.g., .log) remains unchanged.
+  This naming convention ensures easy organization and retrieval of rotated log files based on their creation time.
+
+- -log-file-age
+  The `-log-file-age` flag is an optional time duration setting that defines how long 
+  old backup log files are retained. This flag is used alongside log rotation (enabled 
+  by setting a positive value for -log-file-size) to help manage storage by removing 
+  outdated backup logs.
+
+  Type: Duration
+  
+  Default: 0 (disabled, no automatic deletion of backup files)
+  
+  Usage: When -log-file-age is set to a positive duration (e.g., 24h for 24 hours) 
+  and log rotation is enabled, backup log files will be automatically deleted if 
+  their last modification time is older than the specified duration from the current time.
```

Comment:
```suggestion
  the time when they were created (as encoded in the file name)
  is older than the specified duration from the current time.
```

"creation time" and "modification time" are attributes of the filesystem entry. That "modification time" is not used by the code.